### PR TITLE
Mac: Add first MOOF original disk sets and start splitting software lists

### PR DESCRIPTION
**Pull request by Firehawke**
New working software list additions (mac_flop_orig.xml)
-------------------------------------------------------

Lode Runner (Version 1.0) [4am, Firehawke]
Balance of Power (Version 1.03) [4am, Firehawke]
Shanghai (Version 1.0) [4am, Firehawke]
Skyfox [4am, Firehawke]
Temple of Apshai Trilogy (Version 1985-09-30) [4am, Firehawke] The Surgeon (Version 1.5) [4am, Firehawke]
Uninvited [4am, Firehawke]
King's Quest (Version 1.10) [4am, Firehawke]
Smash Hit Racquetball (Version 1.01) [4am, Firehawke] The Ancient Art of War [4am, Firehawke]
Hacker II [4am, Firehawke]
Indiana Jones and the Revenge of the Ancients [4am, Firehawke] Rambo: First Blood Part II [4am, Firehawke]
One on One [4am, Firehawke]
Winter Games (Version 1985-10-24) [4am, Firehawke] Winter Games (Version 1985-10-31) [4am, Firehawke] Star Trek: The Kobayashi Alternative (Version 1.0) [4am, Firehawke] Mac Attack [4am, Firehawke]
GATO (Version 1.3) [4am, Firehawke]
Dark Castle (Version 1.0) [4am, Firehawke]
Oids (Version 1.4) [4am, Firehawke]
MacWars [4am, Firehawke]
Shadowgate [4am, Firehawke]
Seven Cities of Gold [4am, Firehawke]
Enchanted Scepters [4am, Firehawke]
Beyond Dark Castle [4am, Firehawke]
Arkanoid (Version 1.00) [4am, Firehawke]
The Chessmaster 2000 (Version 1.02) [4am, Firehawke]

### CODE REVIEW

**Comment by Firehawke** [comment]
Okay, @rb6502, let's hash out anything I screwed up. Nobody commit this until it's been thoroughly vetted, please!

**Comment by ICEknigh7** [comment]
> /* Set up the softlists: clean cracks priority, originals second, others last */

Shouldn't the original unaltered dumps take priority?

**Comment by MooglyGuy** [comment]
> > /* Set up the softlists: clean cracks priority, originals second, others last */
> 
> Shouldn't the original unaltered dumps take priority?

Given that @Firehawke is following the established ordering that has existed in Apple II for years, without you saying anything about it? No.

And given that you've said nothing about the Apple II softlist situation for years, you should probably stay in your lane.

**Comment by Firehawke** [comment]
I had originally pushed for the originals to be the default, but in the case of Apple II it came to light that copy protection on the originals often relied on undocumented behavior of the system that would change between models. There were a number of games that would run just fine on later hardware, except for the copy protection. That was the impetus for making cleanly cracked the default going forward.

**Comment by MooglyGuy** [comment]
Clean-cracked makes sense in this case as well, given the sheer amount of copy protection methods that relied on external factors like code wheels, manuals, and hard-to-copy paper sheets.

**Comment by Firehawke** [comment]
Absolutely valid point! That's another great reason, IMO.

**Comment by galibert** [comment]
Also our current mac emulation is not reliable enough for some of the protections (68000 access timings issues).

**Comment by MooglyGuy** [comment]
@0kmg Given that he's openly promoted and encouraged people in the MSX scene to use 'cooked' tape formats rather than anything that even remotely resembles preservation, I view any of his opinions on how software should be preserved through exactly that lens. I furthermore stand by the fact that since the comment that he quoted is literally *verbatim* from the Apple II code, something he's not once spoken up about, he's simply asking to ask, and hasn't even had the common courtesy to do even the most basic level of homework before speaking up with his ignorant point of view.

Everyone is free to have their opinion, but everyone is equally free to point out the flaws in that opinion. Or do you believe that free speech somehow involves people being able to say whatever ignorant thing they want, and everyone has to listen and have their own free speech suppressed if it might cause you to become butthurt?

**Comment by ICEknigh7** [comment]
Okay, let's try to make things clear once again and see if they stick this time:

- I've always been encouraging everyone to keep their WAVs archived, including people dumping MSX and C64 games. Always, even if you haven't read the conversations I've been having. I've even been looking for a possible way to get a cleaner sound capture than using regular recorders, for an even more definitive way to preserve them if such thing is possible.

- It's not my fault if the people dumping MSX cassettes haven't considered contributing with their non-cooked dumps, after their attempts to just enable the already existing TZX support in the MSX cores were shot down twice. Also, I have nothing to do with them, just showed them some support for the following reasons:

- TZX files are already supported for many systems in MAME and they're not considered "bad dumps".
- Current MSX cassette softlist is 100% actual bad dumps (CAS format only supports standard blocks and doesn't even preserve the pauses between blocks).
- We could have had many of those bad dumps replaced with "best effort" TZX dumps, in a format that's already being used by MAME for literally thousands of games. Now we don't even have those.
- Having TZXs doesn't mean we can't also have WAVs, softlists for other systems have both formats. But we have none for the MSX now.
- And finally, cooked formats are even required to verify the integrity of the WAVs, else there's no way to determine whether they're good or bad dumps since not every loading scheme features checksum checks, and thus an apparently good dump may only show any effects of loading errors late in the game. It would be useful to have both kinds of dumps coexisting.

If there's anything else to be said regarding this, let's take it somewhere else.

**Comment by cuavas** [comment]
This is way off-topic.  If it gets derailed any further, I’m going to delete all the comments and start issuing one-week bans for people to cool off.

**Review by cuavas — CHANGES_REQUESTED** [review]
* You’ve added space indentation.
* Please use lowercase in descriptive text in descriptions for anything that isn’t proper nouns or a verbatim quote.
* Using `release` info elements for the dates dumps were posted is a bad idea.  Its accepted meaning across software lists is the release date of the software, and the convention is to use straight ISO YYMMDD format with no hyphens.

**Comment by Firehawke** [comment]
> * You’ve added space indentation.

Not sure how that keeps happening. Fixed and will be committed shortly.

> * Please use lowercase in descriptive text in descriptions for anything that isn’t proper nouns or a verbatim quote.

`<description>Lode Runner (Version 1.0)</description>` -- I'm not seeing anything here that's problematic, unless you want 'version' to be lowercased across the board.

> * Using `release` info elements for the dates dumps were posted is a bad idea.  Its accepted meaning across software lists is the release date of the software, and the convention is to use straight ISO YYMMDD format with no hyphens.

THIS is a bit of a blindside-- I've been using it as such for almost three years without any complaints about any of it until now. If this is what you're going to insist, I'll strip the release tag from all work I've done and add it back only where I have specific information on release date.

File: hash/mac_flop_orig.xml
Comment (ajrhacker) [inline hash/mac_flop_orig.xml:314]: It seems that `<sharedfeat name="automount" value="no"/>` would be a more proper way of preventing mounting of images at boot time. However, none of the software lists use it currently, and I haven't tested whether it actually works.

Comment (Firehawke) [reply hash/mac_flop_orig.xml:314]: Maybe this is something @rb6502 could implement if it's not in? I'd be willing to go back and tag any non-bootable Apple II and Mac disks with that if that's the best solution, but I'm going to let the driver maintainer make the judgement call on what's best.

Comment (rb6502) [reply hash/mac_flop_orig.xml:314]: Support is there, but as AJR said we have no idea if it works.

Comment (angelosa) [reply hash/mac_flop_orig.xml:314]: I just tried, it will just not mount every item listed in the software node.

Comment (cuavas) [reply hash/mac_flop_orig.xml:314]: But what if you’re using a Mac with multiple drives and trying to mount multiple software list items, like `-flop1 mac_flop_misc:sys605:flop1 -flop2 mac_flop_orig:oidsv14:flop1`?  Wouldn’t disabling automatic mounting of media just confuse the user?

And while I think of it, a guy on reddit was confused by the way the drive names and part names are both `flop` with a number.  Would it make more sense to use `disk` rather than `flop` in part names to make it more apparent that they aren’t the same thing?

**Comment by cuavas** [comment]
> > Please use lowercase in descriptive text in descriptions for anything that isn’t proper nouns or a verbatim quote.
> 
> `<description>Lode Runner (Version 1.0)</description>` -- I'm not seeing anything here that's problematic, unless you want 'version' to be lowercased across the board.

Anything in the description that isn’t part of the title should use lowercase for things that aren’t proper nouns, initialisms, etc. unless it’s a verbatim quote.  @ArcadeShadow has been helping standardise this across software lists.  I know you’ve been using title case in descriptive text in the Apple II software lists, but I’d really appreciate it if this one follows what we‘re standardising on from the start.

> > Using `release` info elements for the dates dumps were posted is a bad idea.  Its accepted meaning across software lists is the release date of the software, and the convention is to use straight ISO YYMMDD format with no hyphens.
> 
> THIS is a bit of a blindside-- I've been using it as such for almost three years without any complaints about any of it until now. If this is what you're going to insist, I'll strip the release tag from all work I've done and add it back only where I have specific information on release date.

You keep pushing huge PRs and merging them yourself before I have time to read them, let alone review them properly.  I only really check for and fix really glaring errors because of the sheer volume. and I’ve kind of given up trying to review them while they’re still PRs because they’re usually right before freeze, and you’ll merge them anyway.

If I’d noticed sooner, I would’ve said something sooner.  But if you’d looked at any other software list, you’d see that it’s used for fine-grained release date when the information is available.  For example, look at `pc98_cd:aitd2` – it has `<info name="release" value="19941203" />` because we know it was released on the 3rd of December, 1994.

> Edit: Propose we instead change all of the info element release to info dump date? I think it's important to tag when these were made when possible, so that we can track if/when fixed versions of a dump come out. Getting actual release dates for 99.9% of this material is virtually impossible and I'd argue that the release tag as you specify it is almost useless as a result but still important to document when we DO have it.

I’m sorry, this just seems absurd.  We have the exact release dates for a lot of software from a variety of sources, for example:
* Press releases and advertisements.
* Trade magazines and product listings.
* Release notes.
* Sources like MobyGames.  For example, see the page for [Lode Runner](https://www.mobygames.com/game/lode-runner/release-info) – the dates are known to the month for quite a few platforms, and to the day for Famicom and PC-98.

Using pc98_cd.xml for example, we have release dates with month or day accuracy for 191 items.  For fmtowns_cd.xml, we have fine-grained release dates for 796 items.

I’d argue that dump dates are more problematic than release dates.  Ideally, dumps should be reproducible, so all good dumps of the same media should be identical (yeah, I know analog media is a thing, tapes stretch, there’s lossy Laserdic CHD compression, and so on).  As such, the software items don’t represent a particular dump of the media – they represent the media itself, and anyone should be able to reproduce the same dump if they have the media.  And if you want to talk about information we don’t have in 99.9% of cases, we have no clue whatsoever when the vast majority of console ROM and CD dumps materialised.

At any rate, “release” is a misleading name for the dump publication date, and it’s already being used for fine-grained release dates in multiple software lists.  If you want to keep the dump publication dates in `info` elements, choose a different name for them that’s more descriptive.  It shouldn’t be hard to change any existing ones that are misusing `release` as they’re isolated to a few software lists.

**Pull request closed without merging**
